src/hazelcore/TPat.re

```diff
@@ -0,0 +1,50 @@
+open Sexplib.Std;
+
+/* type variable pattern errors */
+module HoleReason = {
+  [@deriving sexp]
+  type t =
+    | ReservedKeyword
+    | BuiltinType
+    /* TODO: (eric) rename InvalidName to InvalidText */
+    | InvalidName;
+};
+
+/* type variable pattern hole status */
+module Status = {
+  [@deriving sexp]
+  type t =
+    | NotInHole
+    | InHole(HoleReason.t, MetaVar.t);
+};
+
+[@deriving sexp]
+type t =
+  | EmptyHole
+  | TyVar(Status.t, string);
```

Comment:
use TyVar.t instead of string here, and have InvalidText be its own top-level constructor of this type

Reply:
fixed by e0da4528d